### ISSUE TITLE
Mkdocs no longer respecting set display text for same .md files.

### ISSUE THREAD

**Issue by LizThompson93** [mkdocs/mkdocs#3710]
In the past, what you set in the nav is what you get.

```
nav:
  - Home: 'index.md'
  - 'French Main Dishes': 'French Cuisine.md'
  - 'French Desserts': 'French Cuisine.md'
```

Now it overrides the display text to the first one which is not expected behavior.

![image](https://github.com/mkdocs/mkdocs/assets/96261305/78b312d4-2bda-4d5c-8ea5-1c1518c7018f)

**Comment by pawamoy** [comment]
Seems related to https://github.com/mkdocs/mkdocs/issues/3428#issuecomment-2061111631. Adding the same page multiple times to the navigation is not supported.

**Comment by squidfunk** [comment]
Yup, seems related. However, what I wondered was why the titles are not preserved, which definitely worked before 1.6.

**Comment by LizThompson93** [comment]
> Seems related to [#3428 (comment)](https://github.com/mkdocs/mkdocs/issues/3428#issuecomment-2061111631). Adding the same page multiple times to the navigation is not supported.

But it used to be very recently? I know it did.
